pkg/controllers/user/certsexpiration/certsexpiration.go

```diff
@@ -0,0 +1,169 @@
+package certsexpiration
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/rancher/kontainer-engine/cluster"
+	"github.com/rancher/rancher/pkg/controllers/management/clusterprovisioner"
+	rkecluster "github.com/rancher/rke/cluster"
+	"github.com/rancher/rke/pki"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/cert"
+)
+
+type Controller struct {
+	ClusterLister v3.ClusterLister
+	ClusterClient v3.ClusterInterface
+	ClusterStore  cluster.PersistentStore
+	UserContxt    *config.UserContext
+}
+
+func Register(ctx context.Context, userContext *config.UserContext) {
+	c := &Controller{
+		ClusterLister: userContext.Management.Management.Clusters("").Controller().Lister(),
+		ClusterClient: userContext.Management.Management.Clusters(""),
+		ClusterStore:  clusterprovisioner.NewPersistentStore(userContext.Management.Core.Namespaces(""), userContext.Management.Core),
+		UserContxt:    userContext,
+	}
+
+	userContext.Management.Management.Clusters("").AddHandler(ctx, "certificate-expiration", c.sync)
+}
+
+func (c Controller) sync(key string, cluster *v3.Cluster) (runtime.Object, error) {
+	if key == "" || cluster == nil || cluster.DeletionTimestamp != nil {
```

Comment:
since this is running as user controller, we'll now receive sync event in all clusters, but we need to run only for current cluster.. so we need to add `clusterName` in the Controller struct and check for name, else return. 
example, https://github.com/rancher/rancher/blob/master/pkg/controllers/user/monitoring/clusterHandler.go#L50